dex/pair/src/safe_price.rs

```diff
@@ -100,7 +105,14 @@ pub trait SafePriceModule:
             new_index = (safe_price_current_index % MAX_OBSERVATIONS) + 1;
         }
 
-        if last_price_observation.recording_round == current_round {
+        let rounds_since_last_observation = current_round - last_price_observation.recording_round;
+        let safe_price_round_save_interval_mapper = self.safe_price_round_save_interval();
+        let round_save_interval = match safe_price_round_save_interval_mapper.get() {
+            0 => DEFAULT_ROUND_SAVE_INTERVAL,
+            value => value,
+        };
```

Comment:
You really don't need the mapper, you just care about the value. I'm also not a fan of this code style, avoiding if-else just for the sake of using the cool match.

```suggestion
        let mut round_save_interval = self.safe_price_round_save_interval().get();
        if round_save_interval == 0 {
                round_save_interval = DEFAULT_ROUND_SAVE_INTERVAL;
        }
```

This is way easier to understand imo. Also remember to format after applying suggestion, github editor sucks.

Reply:
You could even go one step further and save `DEFAULT_ROUND_SAVE_INTERVAL` in the `safe_price_round_save_interval` in the init/upgrade functions. That way, you could get rid of this whole logic entirely.

Reply:
Good idea.